// @genius-sports/gs-marketing-ui/dist/index.js
import { AnimatePresence, motion, useReducedMotion } from 'framer-motion';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import { forwardRef, useEffect, createElement } from 'react';

// src/components/Accordion.tsx
function useReducedMotionSafe() {
  return useReducedMotion() ?? false;
}

// src/utils/cn.ts
function cn(...parts) {
  return parts.filter(Boolean).join(" ");
}
function AccordionToggleIcon({ isOpen, className }) {
  return /* @__PURE__ */ jsx("span", { className: cn("block h-6 w-6 shrink-0", className), "aria-hidden": "true", children: /* @__PURE__ */ jsxs("svg", { viewBox: "0 0 24 24", className: "h-6 w-6", children: [
    /* @__PURE__ */ jsx(
      "rect",
      {
        width: "24",
        height: "24",
        rx: "12",
        className: cn(
          "fill-current",
          isOpen ? "text-surface" : "text-[var(--color-accordion-green)]"
        )
      }
    ),
    /* @__PURE__ */ jsx(
      "path",
      {
        d: "M7.5 12h9",
        fill: "none",
        stroke: "currentColor",
        strokeWidth: "1.8",
        strokeLinecap: "round",
        className: isOpen ? "text-navy" : "text-white"
      }
    ),
    !isOpen ? /* @__PURE__ */ jsx(
      "path",
      {
        d: "M12 7.5v9",
        fill: "none",
        stroke: "currentColor",
        strokeWidth: "1.8",
        strokeLinecap: "round",
        className: "text-white"
      }
    ) : null
  ] }) });
}
var accordionTriggerFocusClass = "transition focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-blue-500/80";
function AccordionMotionPanel({
  open,
  panelId,
  children,
  className,
  motionKey = panelId
}) {
  const reducedMotion = useReducedMotionSafe();
  return /* @__PURE__ */ jsx(AnimatePresence, { initial: false, children: open ? /* @__PURE__ */ jsx(
    motion.div,
    {
      id: panelId,
      initial: { height: 0, opacity: 0 },
      animate: { height: "auto", opacity: 1 },
      exit: { height: 0, opacity: 0 },
      transition: {
        duration: reducedMotion ? 0.12 : 0.22,
        ease: "easeOut"
      },
      className: cn("overflow-hidden", className),
      children
    },
    motionKey
  ) : null });
}
var STRIP_ROWS = [
  { top: "0%", originLeft: true, delay: "0ms" },
  { top: "25%", originLeft: false, delay: "120ms" },
  { top: "50%", originLeft: true, delay: "240ms" },
  { top: "75%", originLeft: false, delay: "360ms" }
];
function GeniusStripHoverBg() {
  return /* @__PURE__ */ jsx(Fragment, { children: STRIP_ROWS.map(({ top, originLeft, delay }) => /* @__PURE__ */ jsx(
    "div",
    {
      className: "pointer-events-none absolute left-0 right-0 z-10 h-1/4 overflow-hidden",
      style: { top },
      "aria-hidden": true,
      children: /* @__PURE__ */ jsx(
        "div",
        {
          className: `h-full w-full scale-x-0 bg-blue transition-transform duration-[350ms] ease-out group-hover:scale-x-100 ${originLeft ? "origin-left" : "origin-right"}`,
          style: { transitionDelay: delay }
        }
      )
    },
    top
  )) });
}
var Button = forwardRef(function Button2({ variant, className, children, type = "button", ...props }, ref) {
  if (variant === "primaryPill") {
    return /* @__PURE__ */ jsx(
      "button",
      {
        ref,
        type,
        className: cn(
          "group inline-flex items-center justify-center focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-navy focus-visible:ring-offset-2",
          className
        ),
        ...props,
        children: /* @__PURE__ */ jsxs("div", { className: "relative cursor-pointer overflow-hidden rounded-pill bg-navy", children: [
          /* @__PURE__ */ jsx(GeniusStripHoverBg, {}),
          /* @__PURE__ */ jsx("span", { className: "relative z-20 block rounded-pill px-[36px] py-[15px] text-center text-[17px] font-heading font-medium leading-none text-white transition-colors duration-300 ease-in-out sm:px-12 md:py-[1.0625rem]", children })
        ] })
      }
    );
  }
  if (variant === "secondaryPill") {
    return /* @__PURE__ */ jsx(
      "button",
      {
        ref,
        type,
        className: cn(
          "group inline-flex items-center justify-center focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-navy focus-visible:ring-offset-2",
          className
        ),
        ...props,
        children: /* @__PURE__ */ jsxs("div", { className: "relative cursor-pointer overflow-hidden rounded-pill bg-navy/5", children: [
          /* @__PURE__ */ jsx(GeniusStripHoverBg, {}),
          /* @__PURE__ */ jsx("span", { className: "relative z-20 block rounded-pill px-4 py-2.5 text-center font-heading text-xs font-medium leading-none text-navy transition-colors duration-300 ease-in-out group-hover:text-white sm:px-6 sm:py-3 sm:text-sm md:px-8 md:py-3.5 lg:px-5 lg:py-[0.8rem] lg:text-[1rem]", children })
        ] })
      }
    );
  }
  if (variant === "solidAccent") {
    return /* @__PURE__ */ jsx(
      "button",
      {
        ref,
        type,
        className: cn(
          "inline-flex items-center justify-center rounded-md bg-accent px-5 py-2.5 text-base font-medium text-white transition-colors hover:bg-accent-hover focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-accent/70",
          className
        ),
        ...props,
        children
      }
    );
  }
  if (variant === "ghost") {
    return /* @__PURE__ */ jsx(
      "button",
      {
        ref,
        type,
        className: cn(
          "rounded-md px-2 py-1 text-sm font-medium text-muted transition-colors hover:bg-lightGrey hover:text-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-accent/70",
          className
        ),
        ...props,
        children
      }
    );
  }
  return /* @__PURE__ */ jsx(
    "button",
    {
      ref,
      type,
      className: cn(
        "inline-flex items-center justify-center rounded-md border border-white/30 px-4 py-2 text-sm font-medium text-white transition-colors hover:bg-white/10 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-white/50",
        className
      ),
      ...props,
      children
    }
  );
});
function Card({ variant, className, children, ...props }) {
  if (variant === "brand") {
    return /* @__PURE__ */ jsx(
      "div",
      {
        className: cn(
          "rounded-brand border border-lightGrey bg-lightGrey transition duration-[250ms] ease-in-out",
          "hover:-translate-y-1 hover:shadow-card-hover",
          className
        ),
        ...props,
        children
      }
    );
  }
  if (variant === "feature") {
    return /* @__PURE__ */ jsx(
      "div",
      {
        className: cn("overflow-hidden rounded-brand border border-border bg-white", className),
        ...props,
        children
      }
    );
  }
  return /* @__PURE__ */ jsx("div", { className: cn("rounded-brand border border-border bg-transparent", className), ...props, children });
}
function Chip({ className, children, ...props }) {
  return /* @__PURE__ */ jsx(
    "span",
    {
      className: cn(
        "inline-flex items-center rounded-full border border-border bg-lightGrey px-3 py-1 font-body text-xs text-muted md:text-sm",
        className
      ),
      ...props,
      children
    }
  );
}
var panelMaxClass = {
  md: "max-w-lg",
  xl: "max-w-4xl"
};
function Modal({
  open,
  onClose,
  titleId,
  title,
  children,
  size = "xl",
  closeLabel = "Close",
  panelClassName,
  bodyClassName
}) {
  const reducedMotion = useReducedMotionSafe();
  useEffect(() => {
    if (typeof document === "undefined" || !open) {
      return;
    }
    const previousBodyOverflow = document.body.style.overflow;
    document.body.style.overflow = "hidden";
    const handleKeyDown = (event) => {
      if (event.key === "Escape") {
        onClose();
      }
    };
    window.addEventListener("keydown", handleKeyDown);
    return () => {
      document.body.style.overflow = previousBodyOverflow;
      window.removeEventListener("keydown", handleKeyDown);
    };
  }, [open, onClose]);
  return /* @__PURE__ */ jsx(AnimatePresence, { children: open ? /* @__PURE__ */ jsx(
    motion.div,
    {
      className: "fixed inset-0 z-[100] flex items-center justify-center px-4 py-6",
      style: { backgroundColor: "var(--overlay-backdrop)" },
      role: "dialog",
      "aria-modal": "true",
      "aria-labelledby": titleId,
      initial: { opacity: 0 },
      animate: { opacity: 1 },
      exit: { opacity: 0 },
      transition: { duration: reducedMotion ? 0.1 : 0.2, ease: "easeOut" },
      onClick: onClose,
      children: /* @__PURE__ */ jsxs(
        motion.div,
        {
          className: cn(
            "w-full overflow-hidden rounded-xl bg-white shadow-modal-panel",
            panelMaxClass[size],
            panelClassName
          ),
          initial: reducedMotion ? { opacity: 0 } : { opacity: 0, y: 12 },
          animate: { opacity: 1, y: 0 },
          exit: reducedMotion ? { opacity: 0 } : { opacity: 0, y: 10 },
          transition: { duration: reducedMotion ? 0.1 : 0.2, ease: "easeOut" },
          onClick: (event) => event.stopPropagation(),
          children: [
            /* @__PURE__ */ jsxs("div", { className: "flex items-start justify-between border-b border-lightGrey px-4 py-3 md:px-6 md:py-4", children: [
              /* @__PURE__ */ jsx("h3", { id: titleId, className: "m-0 pr-4 font-heading text-brand-h3 text-foreground", children: title }),
              /* @__PURE__ */ jsx(Button, { type: "button", variant: "ghost", onClick: onClose, children: closeLabel })
            ] }),
            /* @__PURE__ */ jsx("div", { className: cn("px-4 pb-4 pt-4 md:px-6 md:pb-6", bodyClassName), children })
          ]
        }
      )
    }
  ) : null });
}
var maxWidthClass = {
  narrow: "max-w-container-narrow",
  default: "max-w-container-default",
  wide: "max-w-container-wide"
};
function SectionShell({ width, className, children, ...props }) {
  return /* @__PURE__ */ jsx("div", { className: cn(maxWidthClass[width], "mx-auto px-6 lg:px-12", className), ...props, children });
}
var headingStyles = {
  h1: "font-heading font-light text-brand-h1 text-navy",
  h2: "font-heading font-light text-brand-h2 text-navy",
  h3: "font-heading font-medium text-brand-h3 text-navy",
  h4: "font-heading font-medium text-brand-h4 text-navy"
};
function Heading({ level, className, ...props }) {
  return createElement(level, {
    className: cn(headingStyles[level], className),
    ...props
  });
}
var textStyles = {
  body: "font-body text-body text-foreground",
  bodySm: "font-body text-body-sm text-muted",
  lead: "font-body text-lead text-foreground",
  leadLg: "font-body text-lead-lg text-foreground",
  leadXl: "font-body text-lead-xl text-foreground",
  muted: "font-body text-body text-muted"
};
function Text({ variant = "body", as = "p", className, ...props }) {
  return createElement(as, {
    className: cn(textStyles[variant], className),
    ...props
  });
}
function MarketingPageLayout({ className, children, ...props }) {
  return /* @__PURE__ */ jsx("div", { className: cn("min-h-screen bg-gs-bg", className), ...props, children });
}
var Section = forwardRef(function Section2({ className, ...props }, ref) {
  return /* @__PURE__ */ jsx("section", { ref, className: cn(className), ...props });
});

// src/tokens/colors.ts
var colors = {
  navy: "#0d1226",
  blue: "#0000dc",
  purple: "#4337a8",
  orange: "#fa5d00",
  lightGrey: "#f6f7f9",
  snow: "#fafafa",
  white: "#ffffff",
  /** Canonical accent — replaces legacy #1D26FF usages */
  accent: "#0011e1",
  accentHover: "#131bdb",
  gsPrimary700: "#0d1226",
  gsAccent500: "#0000dc",
  gsAccent600: "#0011e1",
  gsNeon: "#00ffcc",
  text: "#0d1226",
  textMuted: "#5b6472",
  border: "#e5e7eb",
  success: "#17b26a",
  heroBg: "#0A1330",
  heroTile: "#0011e1",
  heroTileAlt: "#0014ff",
  seqBg: "#0a0a1a",
  seqCard: "#12122a",
  seqAccent: "#6B4EFF",
  seqLabel: "#8B7FFF",
  proofBarAlt: "#1EC971",
  accordionGreen: "#18C971",
  modalAccent: "#66d29f",
  howitworksAccent: "#7dd3fc",
  geniusBlue: "#0011e1",
  overlayBackdrop: "rgba(13,18,38,0.62)"
};
var tailwindColorLiterals = {
  navy: colors.navy,
  blue: colors.blue,
  purple: colors.purple,
  orange: colors.orange,
  lightGrey: colors.lightGrey,
  snow: colors.snow,
  "gs-primary-700": colors.gsPrimary700,
  "gs-accent-500": colors.gsAccent500,
  "gs-accent-600": colors.gsAccent600,
  "gs-neon": colors.gsNeon,
  text: colors.text,
  "text-muted": colors.textMuted,
  border: colors.border,
  success: colors.success
};

// src/tokens/typography.ts
var fontFamilies = {
  heading: ["ESKlarheitKurrentTRIAL", "system-ui", "sans-serif"],
  body: ["Red Hat Text", "system-ui", "sans-serif"]
};
var fontSizes = {
  "brand-h1": ["4.5rem", { lineHeight: "1.1", letterSpacing: "-0.02em", fontWeight: "300" }],
  "brand-h2": ["2.5rem", { lineHeight: "1.15", letterSpacing: "-0.015em", fontWeight: "300" }],
  "brand-h3": ["1.75rem", { lineHeight: "1.2", letterSpacing: "-0.01em", fontWeight: "500" }],
  "brand-h4": ["1.25rem", { lineHeight: "1.3", letterSpacing: "-0.005em", fontWeight: "500" }],
  body: ["1rem", { lineHeight: "1.6", fontWeight: "400" }],
  "body-sm": ["0.875rem", { lineHeight: "1.5", fontWeight: "400" }],
  "section-title": ["2.5rem", { lineHeight: "1.15", letterSpacing: "-0.015em", fontWeight: "300" }],
  "section-copy": ["1.125rem", { lineHeight: "1.6", letterSpacing: "0", fontWeight: "400" }],
  /** Was numeric `text-18` — class: `text-lead` */
  lead: ["1.125rem", { lineHeight: "1.6", letterSpacing: "0", fontWeight: "400" }],
  /** Was numeric `text-20` — class: `text-lead-lg` */
  "lead-lg": ["1.25rem", { lineHeight: "1.55", letterSpacing: "0", fontWeight: "400" }],
  /** Was numeric `text-22` (hero subhead) — class: `text-lead-xl` */
  "lead-xl": ["1.375rem", { lineHeight: "1.55", letterSpacing: "0", fontWeight: "400" }]
};
var fontWeights = {
  light: "300",
  normal: "400",
  medium: "500",
  semibold: "500",
  bold: "700"
};

// src/tokens/spacing.ts
var spacingScale = {
  1: "0.5rem",
  2: "1rem",
  3: "1.5rem",
  4: "2rem",
  5: "2.5rem",
  6: "3rem"
};
var sectionShellPadding = {
  x: "1.5rem",
  xLg: "3rem"
};
var containerWidthsPx = {
  narrow: 1110,
  default: 1280,
  wide: 1376
};

// src/tokens/radius.ts
var radius = {
  md: "0.5rem",
  brand: "0.5rem",
  pill: "9999px"
};

// src/tokens/shadows.ts
var boxShadows = {
  soft: "0 2px 8px rgba(0,0,0,0.06)",
  card: "0 4px 24px rgba(0,0,0,0.10)",
  cardHover: "0 15px 30px -8px rgba(0,0,0,0.08)",
  sequence: "0 8px 48px rgba(0,0,0,0.32)",
  handle: "0 2px 12px rgba(0,0,0,0.24)",
  /** Shared marketing modals */
  modalPanel: "0 20px 60px rgba(2,6,23,0.35)"
};

// src/tokens/breakpoints.ts
var screens = {
  sm: "640px",
  md: "768px",
  lg: "1024px",
  xl: "1280px",
  "2xl": "1536px"
};

export { AccordionMotionPanel, AccordionToggleIcon, Button, Card, Chip, GeniusStripHoverBg, Heading, MarketingPageLayout, Modal, Section, SectionShell, Text, accordionTriggerFocusClass, boxShadows, colors, containerWidthsPx, fontFamilies, fontSizes, fontWeights, radius, screens, sectionShellPadding, spacingScale, tailwindColorLiterals };
