// @genius-sports/gs-marketing-ui/dist/tailwind-preset.js
// src/tokens/colors.ts
var colors = {
  navy: "#0d1226",
  blue: "#0000dc",
  purple: "#4337a8",
  orange: "#fa5d00",
  lightGrey: "#f6f7f9",
  snow: "#fafafa",
  gsPrimary700: "#0d1226",
  gsAccent500: "#0000dc",
  gsAccent600: "#0011e1",
  gsNeon: "#00ffcc",
  text: "#0d1226",
  textMuted: "#5b6472",
  border: "#e5e7eb",
  success: "#17b26a"};
var tailwindColorLiterals = {
  navy: colors.navy,
  blue: colors.blue,
  purple: colors.purple,
  orange: colors.orange,
  lightGrey: colors.lightGrey,
  snow: colors.snow,
  "gs-primary-700": colors.gsPrimary700,
  "gs-accent-500": colors.gsAccent500,
  "gs-accent-600": colors.gsAccent600,
  "gs-neon": colors.gsNeon,
  text: colors.text,
  "text-muted": colors.textMuted,
  border: colors.border,
  success: colors.success
};

// src/tokens/breakpoints.ts
var screens = {
  sm: "640px",
  md: "768px",
  lg: "1024px",
  xl: "1280px",
  "2xl": "1536px"
};

// src/tokens/typography.ts
var fontFamilies = {
  heading: ["ESKlarheitKurrentTRIAL", "system-ui", "sans-serif"],
  body: ["Red Hat Text", "system-ui", "sans-serif"]
};
var fontSizes = {
  "brand-h1": ["4.5rem", { lineHeight: "1.1", letterSpacing: "-0.02em", fontWeight: "300" }],
  "brand-h2": ["2.5rem", { lineHeight: "1.15", letterSpacing: "-0.015em", fontWeight: "300" }],
  "brand-h3": ["1.75rem", { lineHeight: "1.2", letterSpacing: "-0.01em", fontWeight: "500" }],
  "brand-h4": ["1.25rem", { lineHeight: "1.3", letterSpacing: "-0.005em", fontWeight: "500" }],
  body: ["1rem", { lineHeight: "1.6", fontWeight: "400" }],
  "body-sm": ["0.875rem", { lineHeight: "1.5", fontWeight: "400" }],
  "section-title": ["2.5rem", { lineHeight: "1.15", letterSpacing: "-0.015em", fontWeight: "300" }],
  "section-copy": ["1.125rem", { lineHeight: "1.6", letterSpacing: "0", fontWeight: "400" }],
  /** Was numeric `text-18` — class: `text-lead` */
  lead: ["1.125rem", { lineHeight: "1.6", letterSpacing: "0", fontWeight: "400" }],
  /** Was numeric `text-20` — class: `text-lead-lg` */
  "lead-lg": ["1.25rem", { lineHeight: "1.55", letterSpacing: "0", fontWeight: "400" }],
  /** Was numeric `text-22` (hero subhead) — class: `text-lead-xl` */
  "lead-xl": ["1.375rem", { lineHeight: "1.55", letterSpacing: "0", fontWeight: "400" }]
};
var fontWeights = {
  light: "300",
  normal: "400",
  medium: "500",
  semibold: "500",
  bold: "700"
};

// src/tokens/radius.ts
var radius = {
  brand: "0.5rem",
  pill: "9999px"
};

// src/tokens/shadows.ts
var boxShadows = {
  soft: "0 2px 8px rgba(0,0,0,0.06)",
  card: "0 4px 24px rgba(0,0,0,0.10)",
  cardHover: "0 15px 30px -8px rgba(0,0,0,0.08)",
  sequence: "0 8px 48px rgba(0,0,0,0.32)",
  handle: "0 2px 12px rgba(0,0,0,0.24)",
  /** Shared marketing modals */
  modalPanel: "0 20px 60px rgba(2,6,23,0.35)"
};

// src/tokens/spacing.ts
var containerWidthsPx = {
  narrow: 1110,
  default: 1280,
  wide: 1376
};

// src/tailwind-preset.ts
var { narrow, default: defaultW, wide } = containerWidthsPx;
var gsMarketingUiPreset = {
  theme: {
    container: {
      center: true,
      padding: "1rem",
      screens: {
        sm: "640px",
        md: "768px",
        lg: `${defaultW}px`,
        xl: `${defaultW}px`,
        "2xl": `${defaultW}px`
      }
    },
    extend: {
      screens,
      maxWidth: {
        "container-narrow": `${narrow}px`,
        "container-default": `${defaultW}px`,
        "container-wide": `${wide}px`
      },
      fontFamily: {
        heading: [...fontFamilies.heading],
        body: [...fontFamilies.body]
      },
      fontSize: { ...fontSizes },
      fontWeight: {
        ...fontWeights
      },
      colors: {
        ...tailwindColorLiterals,
        "gs-bg": "var(--gs-bg)",
        "gs-surface": "var(--gs-surface)",
        foreground: "var(--color-foreground)",
        muted: "var(--color-muted)",
        surface: "var(--color-surface)",
        accent: "var(--color-accent)",
        "accent-hover": "var(--color-accent-hover)"
      },
      borderRadius: {
        brand: radius.brand,
        pill: radius.pill
      },
      boxShadow: {
        soft: boxShadows.soft,
        card: boxShadows.card,
        "card-hover": boxShadows.cardHover,
        sequence: boxShadows.sequence,
        handle: boxShadows.handle,
        "modal-panel": boxShadows.modalPanel
      }
    }
  }
};
var tailwind_preset_default = gsMarketingUiPreset;

export { tailwind_preset_default as default, gsMarketingUiPreset };
